File: src/main/java/com/wanderers/hotelier_webservices/server/mapper/AccommodationRowMapper.java
package com.wanderers.hotelier_webservices.server.mapper;

import com.wanderers.hotelier_webservices.server.dto.AccommodationDto;
import com.wanderers.hotelier_webservices.server.dto.Category;
import com.wanderers.hotelier_webservices.server.dto.ReputationBadge;
import org.springframework.jdbc.core.RowMapper;

import java.sql.ResultSet;
import java.sql.SQLException;

/**
 * Class is responsible for mapping a database accommodation entity into a server object
 */
public class AccommodationRowMapper implements RowMapper<AccommodationDto> {

    @Override
    public AccommodationDto mapRow(ResultSet rs, int rowNum) throws SQLException {

        AccommodationDto accommodationRow = new AccommodationDto();

        accommodationRow.setId(Integer.parseInt(rs.getString("id")));
        accommodationRow.setHotelierId(rs.getString("hotelier_id"));
        accommodationRow.setName(rs.getString("name"));
        accommodationRow.setRating(Integer.parseInt(rs.getString("rating")));
        accommodationRow.setCategory(Category.fromValue(rs.getString("category").toLowerCase()));
        accommodationRow.setCity(rs.getString("city"));
        accommodationRow.setState(rs.getString("state"));
        accommodationRow.setCountry(rs.getString("country"));
        accommodationRow.setZipCode(rs.getString("zip_code"));
        accommodationRow.setAddress(rs.getString("address"));
        accommodationRow.setImage(rs.getString("image"));
        accommodationRow.setReputation(Integer.parseInt(rs.getString("reputation")));
        accommodationRow.setReputationBadge(ReputationBadge.fromValue(rs.getString("reputation_badge").toLowerCase()));
        accommodationRow.setPrice(Integer.parseInt(rs.getString("price")));
        accommodationRow.setAvailability(Integer.parseInt(rs.getString("availability")));

        return accommodationRow;
    }
}

File: src/main/kotlin/com/wanderers/hotelier_webservices/server/mapper/AccommodationRowMapper.kt
package com.wanderers.hotelier_webservices.server.mapper

import com.wanderers.hotelier_webservices.rest.model.ReputationBadgeEnum
import com.wanderers.hotelier_webservices.server.dto.AccommodationDto
import org.springframework.jdbc.core.RowMapper
import java.sql.ResultSet

/**
 * Class is responsible for mapping a database accommodation entity into a server object
 */
class AccommodationRowMapper : RowMapper<AccommodationDto> {
    override fun mapRow(rs: ResultSet, rowNum: Int): AccommodationDto {

        return AccommodationDto(
            id = rs.getInt("id"),
            hotelierId = rs.getString("hotelier_id"),
            name = rs.getString("name"),
            rating = rs.getInt("rating"),
            category = rs.getString("category"),
            city = rs.getString("city"),
            state = rs.getString("state"),
            country = rs.getString("country"),
            zipCode = rs.getString("zip_code"),
            address = rs.getString("address"),
            image = rs.getString("image"),
            reputation = rs.getInt("reputation"),
            reputationBadge = ReputationBadgeEnum.valueOf(rs.getString("reputation_badge").lowercase()),
            price = rs.getInt("price"),
            availability = rs.getInt("availability")
        )
    }
}
